File: SalesSystem/SalesAPI/sales/models/sale.py
from django.db import models


class Sale(models.Model):
    sale_name = models.CharField(max_length=100)
    nit = models.IntegerField()
    client = models.ForeignKey('Client', on_delete=models.CASCADE)
    fuel_quantity = models.FloatField()
    total = models.FloatField()
    fuel_price = models.FloatField()
    fuel_type = models.ForeignKey('FuelType', on_delete=models.CASCADE)
    pump = models.ForeignKey('Pump', on_delete=models.CASCADE)
    date = models.DateTimeField(auto_now_add=True)
    is_valid = models.BooleanField(default=True)

File: SalesSystem/Sales/src/models/objects/Sale.ts
import { Client } from "./Client";
import { FuelType } from "./FuelType";
import { Pump } from "./Pump";

export interface Sale {
    id?: number;
    sale_name: string;
    nit: number;
    client?: Client;
    client_id?: number;
    fuel_quantity: number;
    total?: number;
    fuel_price?: number;
    fuel_type?: FuelType;
    fuel_type_id?: number;
    pump?: Pump;
    pump_id?: number;
    date?: string;
    is_valid?: boolean;
}